### PR TITLE
issue-711: widget settings screen

### PR DESCRIPTION
**Pull request by stembitf**

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -153,8 +153,9 @@ android {
 
 dependencies {
     implementation 'androidx.work:work-runtime:2.7.1'
+    implementation 'androidx.appcompat:appcompat:1.6.1'
     androidTestImplementation('com.wix:detox:+')
-    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.appcompat:appcompat:1.7.0'
```
Comment (geosaaga) [inline android/app/build.gradle:158]: androidx.appcompat:appcompat is added twice with different version numbers, one should be enough?

File: android/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -62,4 +62,9 @@
     <string name="location_service_info_title">Do you want to enable location service?</string>
     <string name="ask_permission">Ask permission</string>
     <string name="updated">"Updated"</string>
+    <string name="device_mode">Use device mode</string>
```
Comment (geosaaga) [inline android/app/src/main/res/values/strings.xml:65]: This one seems to be missing fi/sv translations

**Review by geosaaga — CHANGES_REQUESTED** [review]
Couldn't really review this one, because doesn't work in my device or emulator if I start on clean state (meaning app was uninstalled before launch). So I don't have any favorites and I haven't given any permissions to the app.

In Samsung A35 I get to settings screen, but it didn't ask me location permission and widget doesn't appear on home screen.

In emulator it doesn't show even settings screen.

**Comment by geosaaga** [comment]
I'm using Android API level 35 (Android 15) in emulator.

Screenshot_20241210_120232

File: android/app/src/main/res/values/strings.xml (hunk 2 above)
Comment (stembitf) [reply android/app/src/main/res/values/strings.xml:65]: Are these translations OK?:
- Käytä laitetilaa
- Använda enhetsläget

File: android/app/build.gradle (hunk 1 above)
Comment (stembitf) [reply android/app/build.gradle:157]: Fixed

**Comment by stembitf** [comment]
I check this...

> Couldn't really review this one, because doesn't work in my device or emulator if I start on clean state (meaning app was uninstalled before launch). So I don't have any favorites and I haven't given any permissions to the app.
> 
> In Samsung A35 I get to settings screen, but it didn't ask me location permission and widget doesn't appear on home screen.
> 
> In emulator it doesn't show even settings screen.

File: android/app/src/main/res/values/strings.xml (hunk 2 above)
Comment (geosaaga) [reply android/app/src/main/res/values/strings.xml:65]: In Figma the english translation is "Use default". iOS widgets use that one and then

fi: Oletus
sv: Använd standard (suggested by ChatGPT)

Comment (stembitf) [reply android/app/src/main/res/values/strings.xml]: Well, OK, I use that too now.
Changed.

**Comment by stembitf** [comment]
@geosaaga 
I fixed the crash issue and other issues related to updates of multiple widgets.

In emulator I noticed a problem. 
Steps:
- Uninstall and install of the FMI app
- Add a first widget with current location
- -> The widget shows nothing (or error view)

It works OK in my phone however.

File: android/app/src/main/res/values-fi/strings.xml [hunk 3]
```diff
@@ -60,4 +60,9 @@
     <string name="location_service_info_title">Haluatko sallia paikannuksen?</string>
     <string name="ask_permission">Salli paikannus</string>
     <string name="updated">"Päivitetty klo "</string>
+    <string name="device_mode">Oletus</string>
+    <string name="add_your_favorite_locations">Lisää suosikkipaikkasi</string>
+    <string name="add_favorite_locations_explanation">Lisää suosikkipaikkasi Sää-sovellukseen, jos haluat käyttää widgetiä ilman sijaintipalveluita.</string>
+    <string name="add_more_favorite_locations">Lisää suosikkipaikkoja</string>
+    <string name="allow_background_location_service">Anna Sää-sovelluksen taustasijaintipalvelun käyttää nykyistä sijaintiasi näyttääkseen lähelläsi sään.</string>
```
Comment (geosaaga) [inline android/app/src/main/res/values-fi/strings.xml:67]: Teksti on hieman huonoa suomen kieltä, parempi olisi:

Salli taustasijaintipalvelu, jotta widget voi käyttää nykyistä sijaintiasi säätietojen näyttämiseen.

File: android/app/src/main/res/values-fi/strings.xml [hunk 4]
```diff
@@ -60,4 +60,9 @@
     <string name="location_service_info_title">Haluatko sallia paikannuksen?</string>
     <string name="ask_permission">Salli paikannus</string>
     <string name="updated">"Päivitetty klo "</string>
+    <string name="device_mode">Oletus</string>
+    <string name="add_your_favorite_locations">Lisää suosikkipaikkasi</string>
+    <string name="add_favorite_locations_explanation">Lisää suosikkipaikkasi Sää-sovellukseen, jos haluat käyttää widgetiä ilman sijaintipalveluita.</string>
```
Comment (geosaaga) [inline android/app/src/main/res/values-fi/strings.xml:65]: "widgetiä" on kirjoitusvirhe, lisäksi hieman parempi olisi:

Lisää suosikkipaikkasi Sää-sovelluksessa, jos haluat käyttää widgettiä ilman sijaintipalveluita.

**Review by geosaaga — APPROVED** [review]
Otherwise OK, but automatic theme (Use default) doesn't work well when OS theme  changes, because light/dark mode is handled manually. Theme doesn't change immediately, but probably it will change when widget is updated next time. I can create ticket to improve theme handling.

File: android/app/src/main/res/values-fi/strings.xml (hunk 3 above)
Comment (stembitf) [reply android/app/src/main/res/values-fi/strings.xml]: Fixed

File: android/app/src/main/res/values-fi/strings.xml (hunk 4 above)
Comment (stembitf) [reply android/app/src/main/res/values-fi/strings.xml]: Fixed

**Pull request merged**